avocado/core/loader.py

```diff
@@ -694,8 +693,7 @@ def ignore_broken(klass, uid, params=None):
                                                 subtests_filter)
             else:
                 if os.access(test_path, os.X_OK):
-                    return self._make_test(test.SimpleTest,
-                                           pipes.quote(test_path))
+                    return self._make_test(test.SimpleTest, test_path)
```

Comment:
I mean this:

```diff
diff --git a/avocado/core/loader.py b/avocado/core/loader.py
index 97dbf5e..1bc116d 100644
--- a/avocado/core/loader.py
+++ b/avocado/core/loader.py
@@ -12,6 +12,7 @@
 # Copyright: Red Hat Inc. 2014
 # Authors: Lucas Meneghel Rodrigues <lmr@redhat.com>
 #          Ruda Moura <rmoura@redhat.com>
+import pipes
 
 """
 Test loader module.
@@ -693,7 +694,8 @@ class FileLoader(TestLoader):
                                                 subtests_filter)
             else:
                 if os.access(test_path, os.X_OK):
-                    return self._make_test(test.SimpleTest, test_path)
+                    return self._make_test(test.SimpleTest,
+                                           pipes.quote(test_path))
                 else:
                     return make_broken(test.NotATest, test_path)

```

works well, passes the `make check` and I'm able to run both examples above with expected results.